tropopause_find/tropopause_find.F90

```diff
@@ -0,0 +1,1493 @@
+! Created: April, 2009
+!
+! CCPP-ized: Haipeng Lin, August 2024
+module tropopause_find
+  !---------------------------------------------------------------
+  ! ... variables for the tropopause module
+  !---------------------------------------------------------------
+
+  use ccpp_kinds,           only : kind_phys
+
+  implicit none
+
+  private
+
+  ! CCPP-compliant subroutines
+  public :: tropopause_find_init
+  public :: tropopause_find_run
+
+  ! "Wrapped" routine for use in old CAM for backward compatibility.
+  ! Also called by tropopause_find_run driver routine.
+  public :: tropopause_findWithBackup
+
+  ! Switches for tropopause method
+  public :: TROP_ALG_NONE, TROP_ALG_ANALYTIC, TROP_ALG_CLIMATE
+  public :: TROP_ALG_STOBIE, TROP_ALG_HYBSTOB, TROP_ALG_TWMO, TROP_ALG_WMO
+  public :: TROP_ALG_CPP
+  public :: NOTFOUND
+
+  save
+
+  ! These parameters define and enumeration to be used to define the primary
+  ! and backup algorithms to be used with the tropopause_find() method. The
+  ! backup algorithm is meant to provide a solution when the primary algorithm
+  ! fail. The algorithms that can't fail are: TROP_ALG_ANALYTIC, TROP_ALG_CLIMATE
+  ! and TROP_ALG_STOBIE.
+  integer, parameter    :: TROP_ALG_NONE      = 1    ! Don't evaluate
+  integer, parameter    :: TROP_ALG_ANALYTIC  = 2    ! Analytic Expression
+  integer, parameter    :: TROP_ALG_CLIMATE   = 3    ! Climatology
+  integer, parameter    :: TROP_ALG_STOBIE    = 4    ! Stobie Algorithm
+  integer, parameter    :: TROP_ALG_TWMO      = 5    ! WMO Definition, Reichler et al. [2003]
+  integer, parameter    :: TROP_ALG_WMO       = 6    ! WMO Definition
+  integer, parameter    :: TROP_ALG_HYBSTOB   = 7    ! Hybrid Stobie Algorithm
+  integer, parameter    :: TROP_ALG_CPP       = 8    ! Cold Point Parabolic
+  integer, parameter    :: TROP_ALG_CHEMTROP  = 9    ! Chemical tropopause
+  
+  integer, parameter    :: TROP_NALG          = 9    ! Number of Algorithms
+  character,parameter   :: TROP_LETTER(TROP_NALG) = (/ ' ', 'A', 'C', 'S', 'T', 'W', 'H', 'F', 'M' /)
+                                                     ! unique identifier for output, don't use P
+
+  ! These variables should probably be controlled by namelist entries.
+  logical ,parameter    :: output_all         = .False.              ! output tropopause info from all algorithms
+  integer ,parameter    :: default_primary    = TROP_ALG_TWMO        ! default primary algorithm
+  integer ,parameter    :: default_backup     = TROP_ALG_CLIMATE     ! default backup algorithm
+
+  ! Namelist variables
+  character(len=256)    :: tropopause_climo_file = 'trop_climo'      ! absolute filepath of climatology file
+
+  integer, parameter :: NOTFOUND = -1
+
+  real(kind_phys), parameter :: ALPHA  = 0.03_kind_phys
+
+  ! FIXME hplin 8/15/24: fillvalue from cam_history_support. To check how it is used
+  ! and if we can remove it with a generic (NOTFOUND?) value
+  real(kind_phys), parameter :: fillvalue = 1.e36_kind_phys
```

Comment:
1. Thanks for the suggestion, I assume if I define it in `src/data/registry.xml` I'd have to define the value somewhere then mark it as initialized. Where would be the best place to assign this? Or can I assign a default value from the registry xml file?

I was wondering if it would be possible to add `.meta` to `cam_history_support.F90` (https://github.com/ESCOMP/CAM-SIMA/blob/0784bcfc66bfdca27af9627b27253b580ffa6694/src/history/cam_history_support.F90) to pass `[fillvalue]`, similar to `tropopause_read_file` passes the climatology?

2. Would `fill_value_for_diagnostic_output` be an appropriate standard name for this?